src/sql/workbench/parts/connection/browser/connection.contribution.ts

```diff
@@ -55,6 +61,48 @@ actionRegistry.registerWorkbenchAction(
 	AddServerAction.LABEL
 );
 
+CommandsRegistry.registerCommand('azdata.connect',
+	function (accessor, args) {
+		const capabilitiesServices = accessor.get(ICapabilitiesService);
+		const notificationService = accessor.get(INotificationService);
+		const connectionManagementService = accessor.get(IConnectionManagementService);
+		if (args && args.serverName && args.providerName
+			&& (args.authenticationType === integrated
+				|| args.authenticationType === azureMFA
+				|| (args.userName && args.password))) {
+			const profile: azdata.IConnectionProfile = {
+				serverName: args.serverName,
+				databaseName: args.database,
+				authenticationType: args.authenticationType,
+				providerName: args.providerName,
+				connectionName: '',
+				userName: args.userName,
+				password: args.password,
+				savePassword: true,
+				groupFullName: undefined,
+				saveProfile: true,
+				id: undefined,
+				groupId: undefined,
+				options: {}
+			};
+			const connectionProfile = ConnectionProfile.fromIConnectionProfile(capabilitiesServices, profile);
```

Comment:
Does doing this fill in the properties bag correctly for all providers, e.g. PGSQL? Have we tested against that?

Reply:
yes, I tested MSSQL and PGSQL